### PR TITLE
Add baidunetdisk

### PR DESCRIPTION
**Pull request by Voya9e**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/20742)

### CODE REVIEW

File: manifests/b/baidu/baidunetdisk/7.4.3.3/baidu.baidunetdisk.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+PackageIdentifier: baidu.baidunetdisk
+PackageVersion: 7.4.3.3
+PackageLocale: en-US
+Publisher: Baidu
+PublisherUrl: https://www.baidu.com/
+PublisherSupportUrl: http://yun.baidu.com/disk/help
+PrivacyUrl: http://yun.baidu.com/disk/privacy
+Author: Baidu
+PackageName: Baidu Netdisk
```
Comment (OfficialEsco) [inline manifests/b/baidu/baidunetdisk/7.4.3.3/baidu.baidunetdisk.locale.en-US.yaml:12]:
```suggestion
Publisher: 北京度友科技有限公司
PublisherUrl: https://www.baidu.com/
PublisherSupportUrl: http://yun.baidu.com/disk/help
PrivacyUrl: http://yun.baidu.com/disk/privacy
Author: Baidu
PackageName: 百度网盘
```

File: manifests/b/baidu/baidunetdisk/7.4.3.3/baidu.baidunetdisk.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- 2
+InstallModes:
+- interactive
+- silent
+- silentWithProgress
+Installers:
+- InstallerLocale: zh-CN
+  Architecture: x86
+  InstallerType: exe
+  Scope: machine
```
Comment (OfficialEsco) [inline manifests/b/baidu/baidunetdisk/7.4.3.3/baidu.baidunetdisk.installer.yaml:17]:
```suggestion
  InstallerType: nullsoft
  Scope: user
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]

**Comment by OfficialEsco** [comment]
```
7/13/2021 10:04:58 AM
Installer Availability and Hash Validation
Status: Started
Installer: https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe

7/13/2021 10:08:37 AM
Status: Completed
##[error] Installer https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe
##[error] Result: Failed
##vso[task.logissue type=error] Error occurred while downloading installer from https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe. 0x80072EE2.
Error information: 
Http Response: OK
Http Reason Phrase: OK

7/13/2021 10:08:38 AM

****************************************************************************************
Installer Validation
****************************************************************************************
Status: Completed
-----------------------------------------------------
Installer https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe
Result: Download Failed
-----------------------------------------------------
Result: Failed
```

If you close and reopen the PR (same button) it will probably work

**Comment by Meng5in** [comment]
> ```
> 7/13/2021 10:04:58 AM
> Installer Availability and Hash Validation
> Status: Started
> Installer: https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe
> 
> 7/13/2021 10:08:37 AM
> Status: Completed
> ##[error] Installer https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe
> ##[error] Result: Failed
> ##vso[task.logissue type=error] Error occurred while downloading installer from https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe. 0x80072EE2.
> Error information: 
> Http Response: OK
> Http Reason Phrase: OK
> 
> 7/13/2021 10:08:38 AM
> 
> ****************************************************************************************
> Installer Validation
> ****************************************************************************************
> Status: Completed
> -----------------------------------------------------
> Installer https://issuecdn.baidupcs.com/issue/netdisk/yunguanjia/BaiduNetdisk_7.4.3.3.exe
> Result: Download Failed
> -----------------------------------------------------
> Result: Failed
> ```
> 
> If you close and reopen the PR (same button) it will probably work

I'll try it.

**Comment by OfficialEsco** [comment]
click the reopen button below 😁

**Comment by Meng5in** [comment]
There is some problem:

`2021-07-12 23:29:48.491 [CLI ] Starting: 'C:\Users\Mengxin\AppData\Local\Temp\WinGet\baidu.baidunetdisk.7.4.3.3.exe' with arguments ''`
`2021-07-12 23:30:14.771 [CLI ] ShellExecute installer failed: 2`
`2021-07-12 23:30:14.773 [CLI ] Terminating context: 0x8a150006 at E:\BA\98\s\external\pkg\src\AppInstallerCLICore\Workflows\ShellExecuteInstallerHandler.cpp:d8`

I don't what's wrong with it, so I add:
 `InstallerSuccessCodes:`
  `- 2`

Can you fix it ?

**Review by OfficialEsco — APPROVED** [review]
Something here is definitively messed up, also just noticed the version number is wrong, i will update the manifest (also make an zh-CH locale)

**Pull request merged**